src/book/03-guides/02-tools/05-catch-guide.mdx

```diff
@@ -0,0 +1,65 @@
+---
+section: Guides
+chapter: Tools
+title: Catch Getting Started
+description: How to write your first test with catch.
+slug: /guides/tools/catch-guide
+---
+
+[Catch](https://github.com/catchorg/Catch2) is the C++ testing framework which we use for our unit tests. In this guide we will go over the basics of Catch by completing a concrete example. This guide assumes familiarity with C++ and testing concepts. To brush-up on testing, see the [Guide for Writing Tests](/guides/general/writing-tests).
+
+Note that the [Catch docs](https://github.com/catchorg/Catch2/tree/devel/docs) should be the first thing to look at if you're wondering about a specific Catch-ism.
+
+## A Basic Example
+
+Catch test cases have a few components. The most important is the `TEST_CASE(..)` macro, which wraps around groups of associated tests. Inside each `TEST_CASE` scope, you should follow the **AAA** structure, _Arranging_ the data first - both inputs and expected outputs - then _Acting_ by calling the function you're testing, then _Asserting_ that the results match the ground truth. We'll make an example for the following toy utility function:
+
+```cpp
+[[nodiscard]] inline int add(const int a, const int b) noexcept {
+    return a + b;
+}
+```
+
+To test it, we'll need pairs of inputs and their associated outputs. Usually, if there's a lot of data, we'll want to keep the it separate from the test logic, but for this example we'll keep it local. Also note that for utilities like this one, we would want a much more comprehensive set of test cases.
+
+```cpp
+using utility::math::add;
+
+TEST_CASE("Testing integer add utility", "[utility][math][add]") {
+    // Arrange
+    static constexpr ssize_t NUM_TESTS = 5;
+    std::array<std::pair<int, int>, NUM_TESTS> inputs = {{0, 0}, {1, 1}, {-1, -1}, {123000, 456}, {-1000, 1000}};
+    std::array<int, NUM_TESTS> ground_truth = {0, 2, -2, 123456, 0};
+    std::array<int, NUM_TESTS> outputs{};
+    // Act
+    for (ssize_t i = 0; i < NUM_TESTS; ++i) {
+        outputs[i] = add(inputs[i].first, inputs[i].second);
+    }
+    // Assert
+    for (ssize_t i = 0; i < NUM_TESTS; ++i) {
+        INFO("In test case number " << i);
+        INFO("Inputs are (" << inputs[i].first << ", " << inputs[i].second << ")");
+        INFO("Ground truth is " << ground_truth[i]);
+        INFO("Function output is " << outputs[i]);
+        REQUIRE(outputs[i] == ground_truth[i]);
+    }
+}
+```
+
+### Dissecting the Example
+
+As we can see in this example, inside the scope of the `TEST_CASE` is where it all happens. We use `INFO` macros to commentate exactly what is happening for each assertion. You shouldn't worry too much about creating huge, unwieldy logs with `INFO` macros, because Catch only prints the `INFO` for test cases which fail by default.
```

Comment:
```suggestion
As we can see in this example, inside the scope of the `TEST_CASE` is where it all happens. We use `INFO` macros to commentate exactly what is happening for each assertion. You shouldn't worry too much about creating huge, unwieldy logs with `INFO` macros, because by default Catch only prints the `INFO` for test cases which fail.
```